app/models/manageiq/providers/google/cloud_manager/event_catcher/stream.rb

```diff
@@ -35,12 +43,22 @@ def events
         subscription.acknowledge(msgs)
       end
     end
+  rescue Fog::Errors::Error
+    raise ProviderUnreachable, "Error when contacting Google Pubsub for events; this may be a temporary failure."
   end
 
   def get_or_create_subscription(google)
+    # If event catcher is not yet setup, then we'll get a fog error
     google.subscriptions.get(subscription_name) ||
       google.subscriptions.create(:name  => subscription_name,
                                   :topic => topic_name)
+  rescue Fog::Errors::NotFound
+    # Rather than expose the notfound error, we expose our own exception
+    # indicating that the worker thread should back off
+    msg = "Unable to find topic #{topic_name}; this likely is because event"\
+        " support for the Google Cloud Platform is not yet setup. Please see"\
+        " the documentation for instructions."
```

Comment:
I don't know if these are messages are presented to the user?  Does verify_credentials walk any of this code?  If the subscriptions are not yet setup, is the event catcher the only place we'll know?  If we present them to the user, I wonder if we need to localize them eventually.  @durandom Can you check with mzazrivec if so? (don't want to spam his email unless it's needed)